src/Turing.jl

```diff
@@ -31,45 +31,43 @@ const CACHERANGES = 0b01
 
 const DEBUG = Bool(parse(Int, get(ENV, "DEBUG_TURING", "0")))
 
+# Random probability measures.
+include("stdlib/distributions.jl")
+include("stdlib/RandomMeasures.jl")
+
 # Include the interface. Temporary until the interface is moved
 # to MCMCChains. CSP 2019-05-12
 include("interface/Interface.jl")
 using .Interface
 import .Interface: AbstractSampler
 
 """
-    struct Model{pvars, dvars, F, TData, TDefaults}
-        f::F
-        data::TData
-        defaults::TDefaults
-    end
+struct Model{F, Targs <: NamedTuple}
+    f::F
+    args::TData
+end
 
-A `Model` struct with parameter variables `pvars`, data variables `dvars`, inner
-function `f`, `data::NamedTuple` and `defaults::NamedTuple`.
+A `Model` struct with arguments `args` and inner function `f`.
 """
-struct Model{pvars,
-    dvars,
-    F,
-    TData,
-    TDefaults
-} <: Sampleable{VariateForm,ValueSupport} # May need to find better types
+struct Model{F, Targs <: NamedTuple} <: Sampleable{VariateForm,ValueSupport} # May need to find better types
     f::F
-    data::TData
-    defaults::TDefaults
+    args::Targs
 end
-function Model{pvars, dvars}(f::F, data::TD, defaults::TDefaults) where {pvars, dvars, F, TD, TDefaults}
-    return Model{pvars, dvars, F, TD, TDefaults}(f, data, defaults)
-end
-get_pvars(m::Model{params}) where {params} = Tuple(params.types)
-get_dvars(m::Model{params, data}) where {params, data} = Tuple(data.types)
-get_defaults(m::Model) = m.defaults
-@generated function in_pvars(::Val{sym}, ::Model{params}) where {sym, params}
-    return sym in params.types ? :(true) : :(false)
+(model::Model)(vi) = model(vi, SampleFromPrior())
+(model::Model)(vi, spl) = model(vi, spl, DefaultContext())
+(model::Model)(args...; kwargs...) = model.f(args..., model; kwargs...)
+function Base.getproperty(m::Model, f::Symbol)
```

Comment:
I think we should avoid doing this by simplying calling `getmissing(m)` directly.

Reply:
I can do that.